File: src/main/java/com/robohorse/robopojogenerator/generator/postprocessors/CommonJavaPostProcessor.java
package com.robohorse.robopojogenerator.generator.postprocessors;

import com.robohorse.robopojogenerator.generator.common.ClassDecorator;
import com.robohorse.robopojogenerator.generator.consts.ClassTemplate;
import com.robohorse.robopojogenerator.models.ClassItemModel;
import com.robohorse.robopojogenerator.models.FieldModel;
import com.robohorse.robopojogenerator.models.GenerationModel;

import javax.inject.Inject;
import java.util.Map;

/**
 * Created by vadim on 25.09.16.
 */
public class CommonJavaPostProcessor extends JavaPostProcessor {
    @Inject
    public CommonJavaPostProcessor() {
    }

    @Override
    public String proceedClassBody(ClassItemModel classItemModel, GenerationModel generationModel) {
        final StringBuilder classBodyBuilder = new StringBuilder();
        final StringBuilder classMethodBuilder = new StringBuilder();
        final Map<String, ClassDecorator> classFields = classItemModel.getClassFields();

        for (String objectName : classFields.keySet()) {
            final String classItemValue = classFields.get(objectName).getJavaItem();
            final String itemNameFormatted = generateHelper.formatClassField(objectName);
            classBodyBuilder.append(classTemplateHelper.createFiled(
                    new FieldModel.Builder()
                            .setClassType(classItemValue)
                            .setFieldNameFormatted(itemNameFormatted)
                            .setFieldName(objectName)
                            .setAnnotation(classItemModel.getAnnotation())
                            .build()
            ));
            if (generationModel.isUseSetters()) {
                classMethodBuilder.append(ClassTemplate.NEW_LINE);
                classMethodBuilder.append(classTemplateHelper.createSetter(
                        itemNameFormatted,
                        classItemValue));

            }
            if (generationModel.isUseGetters()) {
                classMethodBuilder.append(ClassTemplate.NEW_LINE);
                classMethodBuilder.append(classTemplateHelper.createGetter(
                        itemNameFormatted,
                        classItemValue));
            }
        }
        classBodyBuilder.append(classMethodBuilder);
        return classBodyBuilder.toString();
    }

    @Override
    public String createClassTemplate(ClassItemModel classItemModel, String classBody) {
        return classTemplateHelper.createClassBody(classItemModel, classBody);
    }
}

File: generator/src/main/kotlin/com/robohorse/robopojogenerator/postrocessing/common/CommonJavaPostProcessor.kt
package com.robohorse.robopojogenerator.postrocessing.common

import com.robohorse.robopojogenerator.properties.ClassItem
import com.robohorse.robopojogenerator.properties.templates.ClassTemplate
import com.robohorse.robopojogenerator.utils.ClassGenerateHelper
import com.robohorse.robopojogenerator.utils.ClassTemplateHelper
import com.robohorse.robopojogenerator.models.FieldModel
import com.robohorse.robopojogenerator.models.GenerationModel
import com.robohorse.robopojogenerator.models.Visibility

internal class CommonJavaPostProcessor(
    generateHelper: ClassGenerateHelper,
    classTemplateHelper: ClassTemplateHelper
) : JavaPostProcessor(generateHelper, classTemplateHelper) {

    override fun proceedClassBody(
        classItem: ClassItem,
        generationModel: GenerationModel
    ): String {
        val classBodyBuilder = StringBuilder()
        val classMethodBuilder = StringBuilder()
        val classFields = classItem.classFields
        with(classBodyBuilder) {
            for (objectName in classFields.keys) {
                val classItemValue = classFields[objectName]?.getJavaItem(
                    primitive = generationModel.javaPrimitives
                )
                val itemNameFormatted = generateHelper.formatClassField(objectName)
                append(
                    classTemplateHelper.createField(
                        FieldModel(
                            classType = classItemValue,
                            fieldNameFormatted = itemNameFormatted,
                            fieldName = objectName,
                            annotation = classItem.annotation,
                            visibility = if (generationModel.useLombokValue) Visibility.NONE else Visibility.PRIVATE
                        )
                    )
                )
            }
            for (objectName in classFields.keys) {
                val classItemValue = classFields[objectName]?.getJavaItem(
                    primitive = generationModel.javaPrimitives
                )
                val itemNameFormatted = generateHelper.formatClassField(objectName)
                if (generationModel.useSetters) {
                    append(ClassTemplate.NEW_LINE)
                    append(
                        classTemplateHelper.createSetter(
                            itemNameFormatted,
                            classItemValue
                        )
                    )
                }
                if (generationModel.useGetters) {
                    append(ClassTemplate.NEW_LINE)
                    append(
                        classTemplateHelper.createGetter(
                            itemNameFormatted,
                            classItemValue
                        )
                    )
                }
            }
            if (generationModel.useStrings) {
                append(ClassTemplate.NEW_LINE)
                append(
                    classTemplateHelper.createToString(
                        classItem
                    )
                )
            }
            append(classMethodBuilder)
            return toString()
        }
    }

    override fun createClassTemplate(
        classItem: ClassItem,
        classBody: String?,
        generationModel: GenerationModel
    ) = classTemplateHelper.createClassBody(classItem, classBody)
}
